### PR TITLE
c8d: Skip TestRemoveImageGarbageCollector

### PR DESCRIPTION
**Pull request by rumpl**
**- What I did**

- Skipped `TestRemoveImageGarbageCollector`, it tests the layer store
- Fixed `TestRemoveContainerAfterLiveRestore`, there is no need to pass the storage driver when starting the daemon

**- How I did it**

**- How to verify it**

**- Description for the changelog**

**- A picture of a cute animal (not mandatory but encouraged)**

This test checks how the layer store works, so we don't need it when we
use containerd as image store

Signed-off-by: Djordje Lukic <djordje.lukic@docker.com>

### CODE REVIEW

**Review by neersighted — APPROVED** [review]

**Review by vvoland — APPROVED** [review]

File: integration-cli/docker_cli_daemon_test.go [hunk 1]
```diff
@@ -2540,14 +2540,14 @@ func (s *DockerDaemonSuite) TestExecWithUserAfterLiveRestore(c *testing.T) {
 
 func (s *DockerDaemonSuite) TestRemoveContainerAfterLiveRestore(c *testing.T) {
 	testRequires(c, DaemonIsLinux, overlayFSSupported, testEnv.IsLocalDaemon)
```
Comment (thaJeztah) [inline integration-cli/docker_cli_daemon_test.go:2542]: Is the `overlayFSSupported` still need if we remove the `--storage-driver` option?

**Review by thaJeztah — APPROVED** [review]
LGTM

File: integration-cli/docker_cli_daemon_test.go (hunk 1 above)
Comment (rumpl) [reply integration-cli/docker_cli_daemon_test.go]: Removed ;)

**Comment by thaJeztah** [comment]
Failure on Ubuntu is this one (I think I've seen it fail a few times; looks like it's flaky, perhaps we need a tracking ticket if we don't have one yet);

```
=== Failed
=== FAIL: amd64.integration.daemon TestLiveRestore/volume_references/local_volume_with_mount_options (2.41s)
    daemon_test.go:513: assertion failed: error is not nil: Error response from daemon: remove test-live-restore-volume-references-local: volume has active mounts
        --- FAIL: TestLiveRestore/volume_references/local_volume_with_mount_options (2.41s)

=== FAIL: amd64.integration.daemon TestLiveRestore/volume_references (17.68s)
    --- FAIL: TestLiveRestore/volume_references (17.68s)

=== FAIL: amd64.integration.daemon TestLiveRestore (0.00s)
```

**Comment by rumpl** [comment]
Opened an issue https://github.com/moby/moby/issues/46508

**Pull request merged**